### PR TITLE
Web3-less ERC20 quantity type

### PR DESCRIPTION
**Pull request by D4nte**
Step 2 of #1848

Next: Remove usage of U256 as middle man in some parts of the code.

### CODE REVIEW

**Comment by D4nte** [comment]
> Will review once the PR does not include duplicated commits from other PRs.

Indeed sorry, I just assumed https://github.com/comit-network/comit-rs/pull/1899 would be merged by now!

File: cnd/src/asset/ethereum/erc20.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        let u256 = Erc20Quantity::from_wei(u256);
+        let u32 = Erc20Quantity::from_wei(1000u32);
+
+        assert_eq!(u256, u32)
+    }
+
+    #[test]
+    fn from_one_thousand_in_u32_converts_to_u256() {
+        let quantity = Erc20Quantity::from_wei(1000u32);
+
```
Comment (tcharding) [inline cnd/src/asset/ethereum/erc20.rs]:
```suggestion
```

File: cnd/src/asset/ethereum/ether.rs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        let u256 = Ether::from_wei(u256);
+        let u32 = Ether::from_wei(1000u32);
+
+        assert_eq!(u256, u32)
+    }
+
+    #[test]
+    fn from_one_thousand_in_u32_converts_to_u256() {
+        let ether = Ether::from_wei(1000u32);
+
```
Comment (tcharding) [inline cnd/src/asset/ethereum/ether.rs]:
```suggestion
```

File: cnd/src/http_api/mod.rs [hunk 3]
```diff
@@ -503,10 +504,10 @@ mod tests {
     #[test]
     fn http_asset_serializes_correctly_to_json() {
         let bitcoin = HttpAsset::from(asset::Bitcoin::from_sat(100_000_000));
-        let ether = HttpAsset::from(asset::Ether::from_eth(1.0));
+        let ether = HttpAsset::from(asset::Ether::from_wei(1_000_000_000_000_000_000u64));
```
Comment (tcharding) [inline cnd/src/http_api/mod.rs]: Can we have a cost or comment that indicates that this is one Ether in Wei please, I would not have known this without the diff.

File: cnd/src/asset/ethereum/erc20.rs [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    ethereum::{Address, U256},
+};
+use num::{BigUint, Zero};
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use std::{fmt, str::FromStr};
+
+#[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct Erc20Quantity(BigUint);
+
+impl Erc20Quantity {
```
[the remark this answers is not in the document]
Comment (D4nte) [reply cnd/src/asset/ethereum/erc20.rs]: This is used in `Erc20`. you will use `asset::Erc20` around in the code.

**Comment by D4nte** [comment]
Because bors failed to merge #1899 in a timely fashion, these changes will combined in the next PR.

**Pull request closed without merging**